### ISSUE TITLE
Missing consistent order of Javadoc tags

### ISSUE THREAD

**Issue by scordio** [assertj/assertj#3469]
According to Oracle, the Javadoc [order of tags](https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#orderoftags) should be the following:

- `@author`
- `@version`
- `@param`
- `@return`
- `@exception` / `@throws`
- `@see`
- `@since`
- `@serial`
- `@deprecated`

Other than fixing the existing order, we should have an automated way to enforce it, maybe via Spotless or the `maven-javadoc-plugin`.

**Comment by PhilipIngham** [comment]
Hey! I've started working on this issue on [this PR here](https://github.com/bjss/assertj/pull/2), and have mentioned something on the PR and will mention it here just to see where people stand.

In the Oracle Javadoc, it mentions a convention around multiple `@throws` tags to be in alphabetical order - [order of multiple tags](https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#orderingmultipletags). Is this wanting to be implemented? Perhaps it should be a separate issue if it's wanting to go ahead, but figured I'd get some thoughts on it from more knowledgeable folk than myself before deciding what to do with it.

**Comment by scordio** [comment]
@PhilipIngham thanks for jumping on it!

> In the Oracle Javadoc, it mentions a convention around multiple `@throws` tags to be in alphabetical order - [order of multiple tags](https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#orderingmultipletags). Is this wanting to be implemented?

Sounds sensible to me.

Do you already know how to enforce these rules in an automated way?

**Comment by PhilipIngham** [comment]
> Do you already know how to enforce these rules in an automated way?

Not really, no. At the moment I'm just going through everything manually using regex to fix the stuff that's out of place in the tags, but I'll definitely need some help and/or time to figure out how to get the automation down.

**Comment by scordio** [comment]
I was checking what Checkstyle offers and spotted the following:
* https://checkstyle.org/checks/javadoc/index.html
* https://checkstyle.org/checks/javadoc/atclauseorder.html

**Comment by PhilipIngham** [comment]
Checked this out and looks really useful, I will try and implement it :) Thanks for the assistance
